content/docs/guides/ha_scale/perf-scale.md

```diff
@@ -0,0 +1,101 @@
+
+<p align="center">
+  <img src="/docs/images/perf/mesh-topo.png" width="650"/>
+</p>
+
+Note that since both types of service are in the mesh and permissive mode is disabled, thus each request will be authenticated via mTLS. It means that requests go through the Envoy sidecar in both the client and the server.
+
+## Metrics
+
+The test result includes a lot of metrics collected. A clear definition of each metric can help avoid ambiguity. Below are the metric definitions we will use throughout this documentation:
+
+* Control plane
+    * Peak time CPU/Memory: the maximum CPU/memory usage during the testing process.
+    * Stable time CPU/Memory usage: the average CPU/memory usage when the control plane is off the peak time and becomes stable.
+* Data plane sidecar
+    * Peak time CPU/Memory: the maximum CPU/memory usage among all sidecar containers during the testing process.
+    * Stable time CPU/Memory usage: the average CPU/memory usage of all sidecar containers after they become stable.
+* Request latency: request roundtrip time. P50, P90, P99 and P999 of request latency are measured.
+
+The reason that peak time and stable time are measured individually is because of how the services are deployed. All the echo services will be deployed and added to the mesh at the same time. Therefore, the control plane receives a load spike after the deployment starts. All the Envoy sidecars are connecting to the control plane simutaneously. After that, the control plane becomes much more idle, since there is no more change to the mesh topology.
+
+## Performance
+
+In this test, we focus on the request overhead added by OSM as well as the computation resource consumptions. The RPS level of the load generator is calculated as the desired per pod RPS multiplies the number of server side pods. We test the per pod RPS at 500, 1000, 2000 and 4000 respectively.
+
+### Latency
+
+Below are the request latency at different RPS levels, comparing the scenarios with and without OSM installed.
+
+<p align="center">
+  <img src="/docs/images/perf/wo-osm-latency.png" width="650"/>
+</p>
+
+<p align="center">
+  <img src="/docs/images/perf/w-osm-latency.png" width="650"/>
+</p>
+
+The application without OSM performs quite similar across different RPS levels, although there is an upward trend on the P999 latency. Whereas when OSM is installed, the latency starts to increase drastically when it is above 2000 RPS. Looking at the overhead that OSM adds to the P99 latency, it is less than 2ms for RPS below 2000 and around 14ms and 30ms at 2000 and 4000 RPS respectively.
```

Comment:
```suggestion
The application without OSM performs quite similar across different RPS levels, although there is an upward trend on the 99.9 percentile. Whereas when OSM is installed, the latency starts to increase drastically when it is above 2000 RPS. Looking at the overhead that OSM adds to the P99 latency, it is less than 2ms for RPS below 2000 and around 14ms and 30ms at 2000 and 4000 RPS respectively.
```

Reply:
we also may want to put some of these numbers in a table.

Reply:
+1 on the table; it's easier to grok the specifics in a table layout. In said table, it may also be useful to have a column for the delta latency between calls without OSM and calls with OSM

Reply:
I think people won't be interested in the exact numbers in the charts, as the numbers will change for their own applications. The primary purpose is to show the trend. But I do agree the numbers in the charts are crowded.

In this case, we can add a table that compares the p99 latency in scenarios with and w/o OSM.